### PR TITLE
Adding CAT

### PR DESCRIPTION
**Pull request by taylorpaisie**
Pull Request (PR) checklist:
- [X] Include a description of what is in this pull request in this message.
- [X] The dockerfile successfully builds to a test target for the user creating the PR. (i.e. `docker build --tag samtools:1.15test --target test docker-builds/samtools/1.15` )
- [X] Directory structure as name of the tool in lower case with special characters removed with a subdirectory of the version number (i.e. `spades/3.12.0/Dockerfile`)
   - [X] (optional) All test files are located in same directory as the Dockerfile (i.e. `shigatyper/2.0.1/test.sh`)
- [X] Create a simple container-specific [README.md](https://github.com/StaPH-B/docker-builds/blob/master/.github/workflow-templates/readme-template.md) in the same directory as the Dockerfile (i.e. `spades/3.12.0/README.md`)
   - [ ] If this README is longer than 30 lines, there is an explanation as to why more detail was needed
- [X] Dockerfile includes the recommended [LABELS](https://github.com/StaPH-B/docker-builds/blob/master/dockerfile-template/Dockerfile#L8-L18) 
- [X] Main [README.md](https://github.com/StaPH-B/docker-builds/blob/master/README.md) has been updated to include the tool and/or version of the dockerfile(s) in this PR
- [X] [Program_Licenses.md](https://github.com/StaPH-B/docker-builds/blob/master/Program_Licenses.md) contains the tool(s) used in this PR and has been updated for any missing

### CODE REVIEW

File: CAT/5.3/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+LABEL base.image="ubuntu:focal"
+LABEL dockerfile.version="1"
+LABEL software="CAT"
+LABEL software.version=${CAT_VER}
+LABEL description="CAT: a tool for taxonomic classification of contigs and metagenome-assembled genomes (MAGs)."
+LABEL website="https://github.com/dutilh/CAT"
+LABEL license.url="https://github.com/dutilh/CAT/blob/master/LICENSE.md"
+LABEL maintainer="Taylor K. Paisie"
+LABEL maintainer.email='ltj8@cdc.gov'
```
Comment (erinyoung) [inline CAT/5.3/Dockerfile:18]: Can you move your labels to the app stage? They get lost when you switch away from the builder stage.

**Comment by erinyoung** [comment]
Can you change your subdirectory from 'CAT' to 'cat'? Docker is having issues with the upper case letter.

File: CAT/5.3/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+FROM ubuntu:focal AS app
+ARG CAT_VER
+
+# Copy necessary files from the builder stage
+COPY --from=builder /CAT_pack-${CAT_VER}/ /CAT/
+COPY --from=builder /usr/ /usr/
+
+# Add CAT to PATH
+ENV PATH="${PATH}:/CAT/CAT_pack"
+
```
Comment (erinyoung) [inline CAT/5.3/Dockerfile:55]: Can you add the lines to the end of your `app` stage? This sets the default command (CAT --help) and work directory for containers spun up from the image.

```
CMD CAT --help
WORKDIR /data
```

**Comment by erinyoung** [comment]
I think this is looking great! There's just a few small things that need to happen (unless more errors appear):

1. The 'CAT' subdirectory needs to be renamed to 'cat' (it's a docker thing)
2. CMD and WORKDIR is missing from the app stage

**Comment by taylorpaisie** [comment]
Ok should be fixed now, let me know if you have any other suggestions!

**Comment by erinyoung** [comment]
It looks like the tests worked!

```
#15 [test 3/6] RUN mkdir -p db_tests &&     gzip -d /CAT/tests/data/prepare/small.fa.gz &&     CAT prepare --db_fasta /CAT/tests/data/prepare/small.fa     --acc2tax /CAT/tests/data/prepare/prot2acc.txt     --names /CAT/tests/data/prepare/names.dmp     --nodes /CAT/tests/data/prepare/nodes.dmp     --db_dir db_tests/
#15 0.133 [2024-12-04 16:20:50] WARNING: At least 200GB of memory is recommended for large database construction (e.g. nr). 15.6GB is found on your system. You can try to find a machine with more memory if you run into issues or download preconstructed database files from tbb.bio.uu.nl/bastiaan/CAT_prepare/.
#15 0.133 [2024-12-04 16:20:50] Taxonomy folder /data/test/db_tests/tax is created.
#15 0.133 [2024-12-04 16:20:50] Copying nodes.dmp in taxonomy folder.
#15 0.133 [2024-12-04 16:20:50] Copying names.dmp in taxonomy folder.
#15 0.133 [2024-12-04 16:20:50] Database folder /data/test/db_tests/db is created.
#15 0.133 [2024-12-04 16:20:50] Constructing DIAMOND database 2024-12-04_CAT.dmnd from /CAT/tests/data/prepare/small.fa using 4 cores.
#15 0.133 [2024-12-04 16:20:50] DIAMOND database constructed.
#15 0.133 [2024-12-04 16:20:50] Loading file /CAT/tests/data/prepare/nodes.dmp.
#15 0.133 [2024-12-04 16:20:50] Loading file /CAT/tests/data/prepare/small.fa.
#15 0.133 [2024-12-04 16:20:50] Loading file /CAT/tests/data/prepare/prot2acc.txt.
#15 0.133 [2024-12-04 16:20:50] Finding LCA of all protein accession numbers in fasta headers.
#15 0.133 [2024-12-04 16:20:50] Done! File /data/test/db_tests/db/2024-12-04_CAT.fastaid2LCAtaxid is created. 0 of 4 headers (0.0%) corrected. 0 headers (0.0%) do not have a taxid assigned.
#15 0.133 [2024-12-04 16:20:50] Searching database for taxids with multiple offspring.
#15 0.133 [2024-12-04 16:20:50] Writing /data/test/db_tests/db/2024-12-04_CAT.taxids_with_multiple_offspring.
#15 0.133 
#15 0.133 -----------------
#15 0.133 
#15 0.133 [2024-12-04 16:20:50] CAT prepare is done!
#15 0.133 
#15 0.133 Supply the following arguments to CAT or BAT if you want to use this database:
#15 0.133 -d / --database_folder /data/test/db_tests/db
#15 0.133 -t / --taxonomy_folder /data/test/db_tests/tax
#15 DONE 0.1s

#16 [test 4/6] RUN CAT contigs -c burk_wgs_pos_ctrl.fa     -d db_tests/db     -t db_tests/tax
#16 20.86 # CAT v5.3.
#16 20.86 
#16 20.86 CAT is running. Protein prediction, alignment, and contig classification are carried out.
#16 20.86 Rarw!
#16 20.86 
#16 20.86 Supplied command: /CAT/CAT_pack/CAT contigs -c burk_wgs_pos_ctrl.fa -d db_tests/db -t db_tests/tax
#16 20.86 
#16 20.86 Contigs fasta: ./burk_wgs_pos_ctrl.fa
#16 20.86 Taxonomy folder: ./db_tests/tax/
#16 20.86 Database folder: ./db_tests/db/
#16 20.86 Parameter r: 10
#16 20.86 Parameter f: 0.5
#16 20.86 Log file: ./out.CAT.log
#16 20.86 
#16 20.86 -----------------
#16 20.86 
#16 20.86 Doing some pre-flight checks first.
#16 20.86 [2024-12-04 16:20:50] Prodigal found: Prodigal V2.6.3: February, 2016.
#16 20.86 [2024-12-04 16:20:50] DIAMOND found: diamond version 2.1.9.
#16 20.86 Ready to fly!
#16 20.86 
#16 20.86 -----------------
#16 20.86 
#16 20.86 [2024-12-04 16:20:50] Importing contig names from ./burk_wgs_pos_ctrl.fa.
#16 20.86 [2024-12-04 16:20:50] Running Prodigal for ORF prediction. Files ./out.CAT.predicted_proteins.faa and ./out.CAT.predicted_proteins.gff will be generated. Do not forget to cite Prodigal when using CAT or BAT in your publication.
#16 20.86 [2024-12-04 16:21:11] ORF prediction done!
#16 20.86 [2024-12-04 16:21:11] Parsing ORF file ./out.CAT.predicted_proteins.faa
#16 20.86 [2024-12-04 16:21:11] ORFs found on 596 / 596 contigs (100.00%).
#16 20.86 [2024-12-04 16:21:11] Homology search with DIAMOND is starting. Please be patient. Do not forget to cite DIAMOND when using CAT or BAT in your publication.
#16 20.86 			query: ./out.CAT.predicted_proteins.faa
#16 20.86 			database: ./db_tests/db/2024-12-04_CAT.dmnd
#16 20.86 			mode: fast
#16 20.86 			top: 11
#16 20.86 			no-self-hits: False
#16 20.86 			number of cores: 4
#16 20.86 			block-size (billions of letters): 12.0
#16 20.86 			index-chunks: 1
#16 20.86 			tmpdir: ./
#16 20.86 			compress: 0
#16 20.86 [2024-12-04 16:21:11] Homology search done! File ./out.CAT.alignment.diamond created.
#16 20.86 [2024-12-04 16:21:11] Parsing alignment file ./out.CAT.alignment.diamond.
#16 20.86 [2024-12-04 16:21:11] Loading file ./db_tests/tax/nodes.dmp.
#16 20.86 [2024-12-04 16:21:11] Loading file ./db_tests/db/2024-12-04_CAT.fastaid2LCAtaxid.
#16 20.86 [2024-12-04 16:21:11] Loading file ./db_tests/db/2024-12-04_CAT.taxids_with_multiple_offspring.
#16 20.86 [2024-12-04 16:21:11] CAT is spinning! Files ./out.CAT.contig2classification.txt and ./out.CAT.ORF2LCA.txt are created.
#16 20.86 
#16 20.86 -----------------
#16 20.86 
#16 20.86 [2024-12-04 16:21:11] CAT is done! 2/596 contigs (0.34%) have taxonomy assigned.
#16 DONE 20.9s

#17 [test 5/6] RUN CAT bins -b GN02_MAG_IV_B_1-contigs.fa     -d db_tests/db     -t db_tests/tax
#17 0.926 # CAT v5.3.
#17 0.926 
#17 0.926 BAT is running. Protein prediction, alignment, and bin classification are carried out.
#17 0.926 Rarw!
#17 0.926 
#17 0.926 Supplied command: /CAT/CAT_pack/CAT bins -b GN02_MAG_IV_B_1-contigs.fa -d db_tests/db -t db_tests/tax
#17 0.926 
#17 0.926 Bin fasta: ./GN02_MAG_IV_B_1-contigs.fa
#17 0.926 Taxonomy folder: ./db_tests/tax/
#17 0.926 Database folder: ./db_tests/db/
#17 0.926 Parameter r: 5
#17 0.926 Parameter f: 0.3
#17 0.926 Log file: ./out.BAT.log
#17 0.926 
#17 0.926 -----------------
#17 0.926 
#17 0.926 Doing some pre-flight checks first.
#17 0.926 [2024-12-04 16:21:11] Prodigal found: Prodigal V2.6.3: February, 2016.
#17 0.926 [2024-12-04 16:21:11] DIAMOND found: diamond version 2.1.9.
#17 0.926 Ready to fly!
#17 0.926 
#17 0.926 -----------------
#17 0.926 
#17 0.926 [2024-12-04 16:21:11] Importing contig names from ./GN02_MAG_IV_B_1-contigs.fa.
#17 0.926 [2024-12-04 16:21:11] Writing ./out.BAT.concatenated.fasta.
#17 0.926 [2024-12-04 16:21:11] Running Prodigal for ORF prediction. Files ./out.BAT.concatenated.predicted_proteins.faa and ./out.BAT.concatenated.predicted_proteins.gff will be generated. Do not forget to cite Prodigal when using CAT or BAT in your publication.
#17 0.926 [2024-12-04 16:21:12] ORF prediction done!
#17 0.926 [2024-12-04 16:21:12] Parsing ORF file ./out.BAT.concatenated.predicted_proteins.faa
#17 0.926 [2024-12-04 16:21:12] ORFs found on 119 / 119 contigs (100.00%).
#17 0.926 [2024-12-04 16:21:12] Homology search with DIAMOND is starting. Please be patient. Do not forget to cite DIAMOND when using CAT or BAT in your publication.
#17 0.926 			query: ./out.BAT.concatenated.predicted_proteins.faa
#17 0.926 			database: ./db_tests/db/2024-12-04_CAT.dmnd
#17 0.926 			mode: fast
#17 0.926 			top: 11
#17 0.926 			no-self-hits: False
#17 0.926 			number of cores: 4
#17 0.926 			block-size (billions of letters): 12.0
#17 0.926 			index-chunks: 1
#17 0.926 			tmpdir: ./
#17 0.926 			compress: 0
#17 0.926 [2024-12-04 16:21:12] Homology search done! File ./out.BAT.concatenated.alignment.diamond created.
#17 0.926 [2024-12-04 16:21:12] Parsing alignment file ./out.BAT.concatenated.alignment.diamond.
#17 0.926 [2024-12-04 16:21:12] Loading file ./db_tests/tax/nodes.dmp.
#17 0.926 [2024-12-04 16:21:12] Loading file ./db_tests/db/2024-12-04_CAT.fastaid2LCAtaxid.
#17 0.926 [2024-12-04 16:21:12] Loading file ./db_tests/db/2024-12-04_CAT.taxids_with_multiple_offspring.
#17 0.926 [2024-12-04 16:21:12] BAT is flying! Files ./out.BAT.bin2classification.txt and ./out.BAT.ORF2LCA.txt are created.
#17 0.926 
#17 0.926 -----------------
#17 0.926 
#17 0.926 [2024-12-04 16:21:12] BAT is done! 0/1 bins (0.00%) have taxonomy assigned.
#17 0.926 WARNING: since f is set to smaller than 0.5, one bin may have multiple classifications.
#17 DONE 0.9s
```

**Review by erinyoung — APPROVED** [review]
Thank you for putting this together!!!

I have no other changes to recommend. 

I am going to
1. merge this PR
2. deploy to dockerhub and quay with the tags 5.3 and latest

**Comment by erinyoung** [comment]
You can check the status of the deployment here : https://github.com/StaPH-B/docker-builds/actions/runs/12186058730

Let us know if you run into any issues!!!

**Pull request merged**